docs/ruleset_functions/is_hvac_sys_preheat_fluid_loop_attached_to_boiler.md

```diff
@@ -0,0 +1,25 @@
+# is_hvac_sys_preheat_fluid_loop_attached_to_boiler  
+
+**Description:** Returns TRUE if the fluid loop associated with the preheat system associated with the HVAC system is attached to a boiler. Returns FALSE if this is not the case.   
+
+**Inputs:**  
+- **B-RMR**: To evaluate if the fluid loop associated with the preheat system associated with the HVAC system is attached to a boiler.   
+- **hvac_b.id**: The id of the hvac system to evaluate.  
+
+**Returns:**  
+- **is_hvac_sys_preheat_fluid_loop_attached_to_boiler**: Returns TRUE if the fluid loop associated with the preheat system associated with the HVAC system is attached to a boiler. Returns FALSE if this is not the case.   
+ 
+**Function Call:** None  
+
+## Logic:   
+- Set is_hvac_sys_preheat_fluid_loop_attached_to_boiler equal to FALSE: `is_hvac_sys_preheat_fluid_loop_attached_to_boiler = FALSE`  
+- For each boiler in B_RMR, save boiler to loop boiler dictionary: `for boiler_b in B_RMR.RulesetModelInstance.boilers: loop_boiler_dict[boiler_b.loop].append(boiler_b)`
+- Create an object associate with the preheat_system fluid loop associated with hvac_b: `fluid_loop_b = hvac_b.preheat_system[0].hot_water_loop`
+- Check if the fluid loop type is preheat: `if fluid_loop_b.type = "HEATING":`
```

Comment:
small suggestion to change the description of like 18 to say 'Check if the fluid loop type is heating'
it is likely to confuse the RDS team if the line says 'preheat' but checks for 'heating'